test-infra/src/main/java/com/datastax/oss/driver/api/testinfra/ccm/BaseCcmRule.java

```diff
@@ -88,33 +89,86 @@ public void evaluate() throws Throwable {
         // if the test version exceeds the maximum configured one, fail out.
         CassandraVersion maxVersion = CassandraVersion.parse(cassandraRequirement.max());
 
-        if (maxVersion.compareTo(cassandraVersion) <= 0) {
+        if (maxVersion.compareTo(ccmBridge.getCassandraVersion()) <= 0) {
           return new Statement() {
 
             @Override
-            public void evaluate() throws Throwable {
+            public void evaluate() {
               throw new AssumptionViolatedException(
                   "Test requires C* less than "
                       + maxVersion
                       + " but "
-                      + cassandraVersion
+                      + ccmBridge.getCassandraVersion()
                       + " is configured.  Description: "
                       + cassandraRequirement.description());
             }
           };
         }
       }
     }
+
+    DseRequirement dseRequirement = description.getAnnotation(DseRequirement.class);
+    if (dseRequirement != null) {
+      Optional<CassandraVersion> dseVersionOption = ccmBridge.getDseVersion();
+      if (!dseVersionOption.isPresent()) {
+        throw new AssumptionViolatedException("Test Requires DSE but C* is configured.");
```

Comment:
Why can we throw directly here, but in the other cases we have to build a statement?

Reply:
ah yes, this looks like a bug, will fix.